dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1479,180 +1463,191 @@ synchronized String evaluateInterval() {
+                                     + poolSize + ", activeThreads: " + activeThreads + ", queueDepth: " + queueDepth +
+                                     ", tasks completed: " + deltaCompleted);
                     }
                 }
 
-            } else if (shrinkScore >= shrinkScoreFilterLevel && (shrinkScore - growShrinkDiffFilter) > growScore) {
-                poolAdjustment = -poolDecrement;
-            }
+                setPoolIncrementDecrement(poolSize);
 
-            // Force some random variation into the pool size algorithm
-            poolAdjustment = forceVariation(poolSize, poolAdjustment, deltaCompleted, lowActivity);
+                forecast = currentStats.getMovingAverage();
+                shrinkScore = getShrinkScore(poolSize, forecast, throughput, cpuHigh, lowActivity, systemCpuNA);
+                growScore = getGrowScore(poolSize, forecast, throughput, cpuHigh, lowActivity, systemCpuNA);
 
-            // Format an event level trace point with some useful data
-            if (tc.isEventEnabled()) {
-                Tr.event(tc, "Interval data", toIntervalData(throughput, forecast, deltaCompleted, shrinkScore, growScore,
-                                                             poolSize, poolAdjustment));
-            }
+                // Adjust the poolsize only if one of the scores is both larger than the scoreFilterLevel
+                // and sufficiently larger than the other score. These conditions reduce poolsize fluctuation
+                // which might arise due to a weak or noisy signal from the historical throughput data.
 
-            // Change the pool size and save the result, will check it at start of next control cycle
-            targetPoolSize = adjustPoolSize(poolSize, poolAdjustment);
+                if (growScore >= growScoreFilterLevel && (growScore - growShrinkDiffFilter) > shrinkScore) {
+                    poolAdjustment = poolIncrement;
+                    // with available cpu and a persistently deep queue, grow the pool by more than usual
+                    if (!cpuHigh && !systemCpuNA && deepQueue) {
+                        if (consecutiveDeepQueueCycles > consecutiveDeepQueueThreshold) {
+                            poolAdjustment = Math.min(poolAdjustment * deepQueuePoolIncrementMultiple, maxThreads - poolSize);
+                        }
+                    }
+
+                } else if (shrinkScore >= shrinkScoreFilterLevel && (shrinkScore - growShrinkDiffFilter) > growScore) {
+                    poolAdjustment = -poolDecrement;
+                }
+
+                // Force some random variation into the pool size algorithm
+                poolAdjustment = forceVariation(poolSize, poolAdjustment, deltaCompleted, lowActivity);
+
+                // Change the pool size and save the result, will check it at start of next control cycle
+                targetPoolSize = adjustPoolSize(poolSize, poolAdjustment);
+            }
```

Comment:
```suggestion
            }

            // Format an event level trace point with some useful data
            if (tc.isEventEnabled()) {
                Tr.event(tc, "Interval data", toIntervalData(throughput, forecast, deltaCompleted, shrinkScore, growScore,
                                                             poolSize, poolAdjustment));
            }

```
This needs to be done before the finally because `previousThrouput` is part of the interval data.

Reply:
Done